src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -109,32 +109,50 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
         {
             case 1:
             {
-                genConsumeOperands(node);
-                op1Reg = op1->gtRegNum;
-
                 if (node->OperIsMemoryLoad())
                 {
-                    emit->emitIns_R_AR(ins, simdSize, targetReg, op1Reg, 0);
-                }
-                else if ((category == HW_Category_SIMDScalar) && HWIntrinsicInfo::CopiesUpperBits(intrinsicId))
-                {
-                    emit->emitIns_SIMD_R_R_R(ins, simdSize, targetReg, op1Reg, op1Reg);
-                }
-                else if ((ival != -1) && varTypeIsFloating(baseType))
-                {
-                    assert((ival >= 0) && (ival <= 127));
-                    genHWIntrinsic_R_RM_I(node, ins, (int8_t)ival);
+                    genConsumeAddress(op1);
+                    // Until we improve the handling of addressing modes in the emitter, we'll create a
+                    // temporary GT_IND to generate code with.
+                    GenTreeIndir load = indirForm(node->TypeGet(), op1);
+                    emit->emitInsLoadInd(ins, simdSize, node->gtRegNum, &load);
```

Comment:
The old code was using `emitIns_R_AR`, that one has a `Is4ByteSSEInstruction` check that increases the instruction size by 2. `emitInsLoadInd` does not seem to have such a check.

Reply:
Right - I took a quick look at that when you made that comment earlier, and I wasn't sure it was actually required. I need to take another look.

Reply:
That 4 byte check appeared in #16249 and was updated in #21528. It's not clear to me why that was done in `emitIns` functions instead of `emitInsSize` functions.

Reply:
So far I believe that this 2-byte increment should actually be 1 byte. That said, there is definitely inconsistency in how the sizes are determined, and the code allows the size to be over-estimated, so those errors go undetected. I'm trying to decide whether to make a more general cleanup here or just go with the 1 byte increment for this case.

Reply:
I thought we had a work item tracking the cleanup of `emitInsSize`, but I can't seem to find it.

From what I remember, these one offs were done because it was non-trivial to update the existing `emitInsSize` checks due to other complications at the time. I think it is `+2` to handle the non-VEX encoded case but only needs to be `+1` for the VEX encoding.

Reply:
I believe it's the case that for these addressing-mode instructions, the SSE encodings require one additional bit (and the VEX encodings don't require an additional bytes). I've run all the hw intrinsic tests without AVX & AVX2 with the most recent set of changes.
The size analysis is so convoluted that I am mostly relying on empirical results (i.e. testing).

Reply:
I added a note to https://github.com/dotnet/coreclr/issues/23006 that this should also be addressed.